.github/workflows/speech-train-data-ci-cd.yml

```diff
@@ -246,39 +244,42 @@ jobs:
       - name: Create Custom Speech model with training data
         run: |
-          speech model create -l ${{ env.SPEECH_LOCALE }} ${{ env.CUSTOM_SPEECH_MODEL_DATA }} -s ${{ env.BASE_MODEL_ID }} -n custom_speech_model_${{ env.CURRENT_COMMIT_HASH }} --wait > ${{ env.TRAIN_BUILD_FOLDER_PATH }}/model-creation-output.txt
-          custom_speech_model_id=$(cat ${{ env.TRAIN_BUILD_FOLDER_PATH }}/model-creation-output.txt | sed -n '3p')
+          spx csr model create --name custom_speech_model_${CURRENT_COMMIT_HASH} --datasets @my.training.datasets --output url @my.model.url --wait
+          custom_speech_model_id=$(cat my.model.url | tail -c 36)
           if ! [[ ${custom_speech_model_id//-/} =~ ^[[:xdigit:]]{32}$ ]]
           then
             echo "::error ::Failed to create a Custom Speech model. Possibly re-run all jobs."
             exit 1
           fi
-          echo "::set-env name=CUSTOM_SPEECH_MODEL_ID::$(echo $custom_speech_model_id)"
+          echo "CUSTOM_SPEECH_MODEL_ID=$(echo $custom_speech_model_id)" >> $GITHUB_ENV
 
       # Set environment variables to be bash variables to avoid substitution
       # errors. Not all training data will be used, so check that it was with
       # [[${my_variable//-/} =~ ^[[:xdigit:]]{32}$]] which will return true if
       # the variable is a valid GUID with 32 hexadecimal characters.
       - name: Delete training datasets
         run: |
-          audio_trans_train_id=${{ env.AUDIO_TRANS_TRAIN_ID }}
-          if [[ ${audio_trans_train_id//-/} =~ ^[[:xdigit:]]{32}$ ]]
+          datasets_file=my.training.datasets
+          number_of_datasets_to_delete=$( wc -l < "$datasets_file" )
+          if [[ $number_of_datasets_to_delete -gt 0 ]]
           then
-            speech dataset delete ${{ env.AUDIO_TRANS_TRAIN_ID }}
-            echo DELETED AUDIO+HUMAN-LABELED TRANSCRIPT TRAINING DATA.
-          fi
-          pronunciation_id=${{ env.PRONUNCIATION_ID }}
-          if [[ ${pronunciation_id//-/} =~ ^[[:xdigit:]]{32}$ ]]
-          then
-            speech dataset delete ${{ env.PRONUNCIATION_ID }}
-            echo DELETED PRONUNCIATION DATA.
-          fi
-          language_id=${{ env.LANGUAGE_ID }}
-          if [[ ${language_id//-/} =~ ^[[:xdigit:]]{32}$ ]]
-          then
-            speech dataset delete ${{ env.LANGUAGE_ID }}
-            echo DELETED LANGUAGE DATA.
+            while read -a line
+            do
+              line_url=$(cat line | tr -d '[:space:]')
+              line_id=$(cat line_url | tail -c 36)
+              if [[ ${line_id//-/} =~ ^[[:xdigit:]]{32}$ ]]
+              then
+                spx csr dataset delete --dataset $line_url
+                if [[ ${?} -gt 0 ]]
+                then
+                  echo "::warning::Bad deletion request detected"
+                else
+                  echo DELETED DATASET WITH GUID: "$line"
+                fi
+              fi
+            done < "$datasets_file" | head -$number_of_datasets_to_delete
```

Comment:
Great change, it's so much nicer to programmatically work with `@my.training.datasets` than individually referencing dataset guids

Reply:
Thanks! :-)